util/awsservice/cloudwatchmetrics.go

```diff
@@ -57,10 +57,18 @@ func ValidateMetric(metricName, namespace string, dimensionsFilter []types.Dimen
 }
 
 // ValidateMetrics takes the metric name, metric dimension and corresponding namespace that contains the metric
-func ValidateMetricWithTest(t *testing.T, metricName, namespace string, dimensionsFilter []types.DimensionFilter) {
-	err := ValidateMetric(metricName, namespace, dimensionsFilter)
+func ValidateMetricWithTest(t *testing.T, metricName, namespace string, dimensionsFilter []types.DimensionFilter, retries int, retryTime time.Duration) {
+	var err error
+	for i := 0; i < retries; i++ {
+		err = ValidateMetric(metricName, namespace, dimensionsFilter)
```

Comment:
nit: You could log that retry `i` variable so it is a bit clearer in the logs that retries are occurring.
Or maybe it is already  sufficiently clear in the logs.